### DISCUSSION TITLE
How to assign default value of a model to an empty dict?

### DISCUSSION THREAD

**Discussion by mohammadali95** [pydantic/pydantic#8062] [Question] [answered]
```
class Address(BaseModel):
    ZipCode: str

class Phones(BaseModel):
    Type: Optional[Literal['work', 'cell', 'home']] = ''
    AllowText: Optional[str] = "False"
    AllowAutoDial: Optional[str] = "False"
    MobileCarrier: Optional[str] = ''

class SavePersonInputDataModel(BaseModel):
    
    PersonHashId: str
    UserName: str
    FirstName: str
    LastName: str
    Type: owner_types
    Address: Address
    Phones: Phones = None


    @field_validator('*', mode='before')
    def person_hash_exists(cls, v):
        if v == '':
            return None
        return v
    
    @field_validator('Phones', mode='before')
    @classmethod
    def phone_exists(cls, v):
        if v is None:
            return {}
        return v
```

`input = {'PersonHashId' : '213123dasd', 'UserName': 'ben', 'FirstName': 'ali', 'LastName': 'Mohammad', 'Type': 'ncar', 'Address': {'ZipCode':'asd'}}`

The above input returns this object:
`{'PersonHashId': '213123dasd', 'UserName': 'ben', 'FirstName': 'ali', 'LastName': 'Mohammad', 'Type': 'ncar', 'Address': {'ZipCode': 'asd'}, 'Phones': None}`

However I want the 'Phones' key in my object to have the value of an empty dict instead of None i.e 
`{'PersonHashId': '213123dasd', 'UserName': 'ben', 'FirstName': 'ali', 'LastName': 'Mohammad', 'Type': 'ncar', 'Address': {'ZipCode': 'asd'}, 'Phones': {}}`

Is that possible?

**Comment by sydney-runkle** [comment, marked as answer]
Hi @mohammadali95,

Sure thing! You can use a `Field` with `default_factory` set to `dict`:

```py
from pydantic import BaseModel, Field


class Model(BaseModel):
    foo: str = Field(default_factory=dict)

m = Model()
print(repr(m))
#> Model(foo={})
```

**Reply by mohammadali95** [reply]
Thank you @sydney-runkle !
